internal/terraform/context_validate_test.go

```diff
@@ -3571,3 +3571,84 @@ func TestContext2Validate_queryList(t *testing.T) {
 		})
 	}
 }
+
+// Action Validation is largely exercised in context_plan_actions_test.go
+func TestContext2Validate_action(t *testing.T) {
+	tests := map[string]struct {
+		config  string
+		wantErr string
+	}{
+		"valid config": {
+			`
+terraform {
+	required_providers {
+		test = {
+			source = "hashicorp/test"
+			version = "1.0.0"
+		}
+	}
+}
+action "test_register" "foo" {
+  config {
+      host = "cmdb.snot"
+  }
+}
+resource "test_instance" "foo" {
+  lifecycle {
+    action_trigger {
+      events = [after_create]
+      actions = [action.test_register.foo]
+    }
+  }
+}
+`,
+			"",
+		},
+	}
```

Comment:
Can we add a test case where the provider returns a diagnostic and/or an assertion that we call the validation RPC?